clients/python-wrapper/lakefs/client.py

```diff
@@ -0,0 +1,117 @@
+import base64
+import binascii
+from typing import Optional
+import requests
```

Comment:
so no stdlib? why not just use `urllib3` which `lakefs_sdk` [already depends on](https://github.com/treeverse/lakeFS/blob/master/clients/python/setup.py#L27)?

Reply:
IMHO requests is much more superior than urllib3. It provides much more builtin functionality such as modeling of auth types, validation of resp etc.

Reply:
We are using the http library for the upload (create) flow. We are not using the upload_object API as we want to refrain from writing data to file

Reply:
[not good enough](https://urllib3.readthedocs.io/en/stable/advanced-usage.html#streaming-and-i-o)? :)

Reply:
Some examples:
- urllib requires passing method as string while requests implements a function per http method
- urllib supports providing basic_auth key as part of the make_headers method but does not model a general Auth class which can be expanded. And does not merge headers (AFAICT)
- requests has a neat functionality on response object of raise_for_status() which raises a nicely formatted exception if status code != 2xx